### ISSUE TITLE
Can't connect to EAP network

### ISSUE THREAD

**Issue by devenblake** [J-Lentz/iwgtk#9]
OS: `Arch latest`; hardware: `Thinkpad T420 (Intel Corporation Centrino Advanced-N 6205)`; window manager: `i3-gaps`
I'm trying to connect to my school's EAP access point and nothing happens upon clicking the Connect button next to the EAP network. The behavior I expected was for it to provide feedback saying the EAP network was unreachable or provide a way for me to input my access credentials.

Thanks in advance for any help.

**Comment by J-Lentz** [comment]
As far as I know, iwd currently does not support connecting to EAP networks through its D-Bus interface, which means the user must write a network configuration file (See: `man iwd.network`) and place it in /var/lib/iwd/.

Perhaps iwgtk should show a pop-up window containing instructions to write a network configuration file when the user attempts connect to an unknown EAP network. The current behavior (printing an error to stderr and otherwise doing nothing) is certainly less than ideal.

Just to confirm, if you run iwgtk from the command line and attempt to connect to the network, do you see a `net.connman.iwd.NotSupported` error?

**Comment by devenblake** [comment]
Let me try again tomorrow and copy the errors. That functionality would be greatly appreciated! I spent a while today trying to figure out writing a configuration file and just gave up and wired in.

**Comment by devenblake** [comment]
I'm not getting `net.connman.iwd.NotSupported` errors, I seem to actually be getting a combination of `net.connman.iwd.InProgress`es and `(iwgtk:4559): GLib-GIO-CRITICAL **: 11:00:30.487: g_bus_watch_name_on_connection: assertion 'G_IS_DBUS_CONNECTION (connection)' failed`. Every time I try to connect it's one or both of them.

**Comment by efetropy** [comment]
As @J-Lentz  mentioned in his post above, you need to create a configuration file in _/var/lib/iwd_ and save it as _ssid.8021x_
You can find way more information if you look for "iwd" in the ArchWiki. There is also some example configuration which you can adjust to your needs.

I can confirm that using the latter with iwgtk, I have no issues connecting to e.g. eduroam (PEAP/MSCHAPV2), but keep in mind that certain kernels don't play well. I had a lot of issues with kernel 5.8.5, but the newer versions seem to work well. Currently I am on 5.8.14.

**Comment by devenblake** [comment]
Ideally there should at least be a "Function unsupported" message.
